### DISCUSSION TITLE
Program halted after dishwasher door is opened and closed

### DISCUSSION THREAD

**Discussion by fitur** [thoukydides/homebridge-homeconnect#173] [General]
Hi,
I usually open the dishwasher door during a program in case I badly need something that is being cleaned. Whenever I close the dishwasher door again, the program continues within a couple of minutes. 
Ever since I connected the dishwasher to HomeBridge via this plugin, the program is halted indefinitely and I am required to restart it. 
Am I missing something or have I misconfigured something in HomeBridge or HomeKit?

**Comment by thoukydides** [comment]
That shouldn’t happen. I can’t think of any reason why it would unless you have configured a HomeKit automation. It doesn’t do that with my own dishwasher - I regularly open the door in the middle of a programme and it always resumes automatically.

Please provide a Homebridge log file, with debug enabled (using the `-D` command line option). As a minimum this should cover the period around when you open and close the door, but ideally should be a full log from when Homebridge is started.

**Comment by fitur** [comment]
I changed from individual switches to none and the program continued. Must have been a misconfiguration.

>[05/11/2023, 21:19:10] [HomeConnect] Attempting authorisation
>[05/11/2023, 21:19:12] Homebridge v1.6.1 (HAP v0.11.1) (Homebridge 672E) is running on port 51575.
>[05/11/2023, 21:19:12] [HomeConnect] Using saved access token
>[05/11/2023, 21:19:12] [HomeConnect] Successfully authorised
>[05/11/2023, 21:19:12] [HomeConnect] Home Connect request #1: GET /api/homeappliances
>[05/11/2023, 21:19:17] [HomeConnect] Scheduling configuration schema write
>[05/11/2023, 21:19:22] [HomeConnect] Home Connect request #1: 200 OK +9953ms
>[05/11/2023, 21:19:22] [HomeConnect] Found 1 appliances
>[05/11/2023, 21:19:22] [HomeConnect] Connecting accessory 'Diskmaskin'
>[05/11/2023, 21:19:22] [HomeConnect] Starting events stream for all appliances
>[05/11/2023, 21:19:22] [HomeConnect] [Diskmaskin] Bosch Dishwasher (E-Nr: SMU4HAW48S/38)
>[05/11/2023, 21:19:23] [HomeConnect] Home Connect request #2: GET /api/homeappliances/events
>[05/11/2023, 21:19:24] [HomeConnect] [Diskmaskin] Connected, so reading appliance state...
>[05/11/2023, 21:19:24] [HomeConnect] Scheduling overlapping configuration schema write
>[05/11/2023, 21:19:24] [HomeConnect] Home Connect request #3: GET /api/homeappliances/DISHWASHER-ID
>[05/11/2023, 21:19:24] [HomeConnect] [Diskmaskin] Cache loaded (21 entries)
>[05/11/2023, 21:19:24] [HomeConnect] [Diskmaskin] Cache "Appliance power" [sv-SE, updated 2 days 10 hours ago] is too old
>[05/11/2023, 21:19:24] [HomeConnect] [Diskmaskin] Cache "Appliance commands" [sv-SE, updated 2 days 10 hours ago] is too old
>[05/11/2023, 21:19:24] [HomeConnect] [Diskmaskin] Cache "Appliance settings" [sv-SE, updated 2 days 10 hours ago] is too old
>[05/11/2023, 21:19:24] [HomeConnect] [Diskmaskin] Cache "Appliance settings" [sv-SE, updated 2 days 10 hours ago] is too old
>[05/11/2023, 21:19:24] [HomeConnect] [Diskmaskin] Restored details of 8 programs
>[05/11/2023, 21:19:26] [HomeConnect] Home Connect request #2: 200 OK +3833ms
>[05/11/2023, 21:19:26] [HomeConnect] Started events stream for all appliances after 4 seconds 173 milliseconds
>[05/11/2023, 21:19:26] [HomeConnect] [Diskmaskin] Event START (0 items)
>[05/11/2023, 21:19:27] [HomeConnect] Home Connect request #3: 200 OK +2779ms
>[05/11/2023, 21:19:27] [HomeConnect] Home Connect request #4: GET /api/homeappliances/DISHWASHER-ID/status
>[05/11/2023, 21:19:28] [HomeConnect] Scheduled configuration schema write complete
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] Event EVENT (2 items)
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] BSH.Common.Event.ProgramAborted=BSH.Common.EnumType.EventPresentState.Present (2 listeners)
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] Event Program Aborted
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] Dishcare.Dishwasher.Event.SaltNearlyEmpty=BSH.Common.EnumType.EventPresentState.Present (1 listeners)
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] Event Salt Low
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] No program running
>[05/11/2023, 21:19:28] [HomeConnect] Home Connect request #4: 200 OK +699ms
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] BSH.Common.Status.RemoteControlStartAllowed=true (1 listeners)
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] BSH.Common.Status.RemoteControlActive=true (1 listeners)
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] BSH.Common.Status.DoorState=BSH.Common.EnumType.DoorState.Closed (1 listeners)
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] Door closed
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] BSH.Common.Status.OperationState=BSH.Common.EnumType.OperationState.Run (3 listeners)
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] Operation state implies power is on
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] BSH.Common.Setting.PowerState=BSH.Common.EnumType.PowerState.On (1 listeners)
>[05/11/2023, 21:19:28] [HomeConnect] Home Connect request #5: GET /api/homeappliances/DISHWASHER-ID/settings
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] Remote operation enabled (remote control activated, remote start allowed)
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] On
>[05/11/2023, 21:19:28] [HomeConnect] [Diskmaskin] Active (Run)
>[05/11/2023, 21:19:29] [HomeConnect] Home Connect request #5: 200 OK +699ms
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] BSH.Common.Setting.PowerState=BSH.Common.EnumType.PowerState.On (1 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] BSH.Common.Setting.ChildLock=false (0 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] Home Connect request #6: GET /api/homeappliances/DISHWASHER-ID/programs/selected
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] On
>[05/11/2023, 21:19:29] [HomeConnect] Home Connect request #6: 200 OK +345ms
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] BSH.Common.Root.SelectedProgram=Dishcare.Dishwasher.Program.NightWash (0 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] Home Connect request #7: GET /api/homeappliances/DISHWASHER-ID/programs/active
>[05/11/2023, 21:19:29] [HomeConnect] Home Connect request #7: 200 OK +301ms
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] BSH.Common.Root.ActiveProgram=Dishcare.Dishwasher.Program.NightWash (0 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] BSH.Common.Option.StartInRelative=0 seconds (1 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] Dishcare.Dishwasher.Option.ExtraDry=false (0 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] Dishcare.Dishwasher.Option.HalfLoad=false (0 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] Dishcare.Dishwasher.Option.VarioSpeedPlus=false (0 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] Dishcare.Dishwasher.Option.SilenceOnDemand=false (0 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] BSH.Common.Option.ProgramProgress=11 % (0 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] BSH.Common.Option.RemainingProgramTime=12838 seconds (1 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] BSH.Common.Option.RemainingProgramTimeIsEstimated=true (0 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] BSH.Common.Option.EnergyForecast=58 % (0 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] BSH.Common.Option.WaterForecast=52 % (0 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] Successfully read all appliance state
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] connected=true (8 listeners)
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] Connected
>[05/11/2023, 21:19:29] [HomeConnect] [Diskmaskin] Cache "Appliance programs" [sv-SE, updated 2 days 10 hours ago] is too old
>[05/11/2023, 21:19:29] [HomeConnect] Home Connect request #8: GET /api/homeappliances/DISHWASHER-ID/settings/BSH.Common.Setting.PowerState
>[05/11/2023, 21:19:29] [HomeConnect] Home Connect request #9: GET /api/homeappliances/DISHWASHER-ID/commands
>[05/11/2023, 21:19:30] [HomeConnect] Home Connect request #10: GET /api/homeappliances/DISHWASHER-ID/settings
>[05/11/2023, 21:19:30] [HomeConnect] Home Connect request #11: GET /api/homeappliances/DISHWASHER-ID/settings
>[05/11/2023, 21:19:30] [HomeConnect] Home Connect request #12: GET /api/homeappliances/DISHWASHER-ID/programs
>[05/11/2023, 21:19:30] [HomeConnect] [Diskmaskin] Program has 3 hours 33 minutes remaining
>[05/11/2023, 21:19:30] [HomeConnect] [Diskmaskin] On
>[05/11/2023, 21:19:30] [HomeConnect] [Diskmaskin] Active (Run)
>[05/11/2023, 21:19:30] [HomeConnect] Home Connect request #9: 200 OK +788ms
>[05/11/2023, 21:19:30] [HomeConnect] [Diskmaskin] "Appliance commands" written to cache
>[05/11/2023, 21:19:30] [HomeConnect] Home Connect request #8: 200 OK +846ms
>[05/11/2023, 21:19:30] [HomeConnect] [Diskmaskin] BSH.Common.Setting.PowerState=BSH.Common.EnumType.PowerState.On (1 listeners)
>[05/11/2023, 21:19:30] [HomeConnect] [Diskmaskin] "Appliance power" written to cache
>[05/11/2023, 21:19:30] [HomeConnect] Home Connect request #12: 200 OK +693ms
>[05/11/2023, 21:19:30] [HomeConnect] [Diskmaskin] "Appliance programs" written to cache
>[05/11/2023, 21:19:30] [HomeConnect] [Diskmaskin] On
>[05/11/2023, 21:19:31] [HomeConnect] Home Connect request #11: 200 OK +959ms
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] BSH.Common.Setting.PowerState=BSH.Common.EnumType.PowerState.On (1 listeners)
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] BSH.Common.Setting.ChildLock=false (0 listeners)
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] "Appliance settings" written to cache
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] On
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] Cache saved (21 entries)
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] Can be switched off
>[05/11/2023, 21:19:31] [HomeConnect] Home Connect request #13: GET /api/homeappliances/DISHWASHER-ID/programs/available
>[05/11/2023, 21:19:31] [HomeConnect] Home Connect request #10: 200 OK +1125ms
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] BSH.Common.Setting.PowerState=BSH.Common.EnumType.PowerState.On (1 listeners)
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] BSH.Common.Setting.ChildLock=false (0 listeners)
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] "Appliance settings" written to cache
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] On
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] Cache saved (21 entries)
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] Does not support Functional Light
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] Cache "Appliance settings" [sv-SE, updated 83 milliseconds ago]
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] Does not support Ambient Light
>[05/11/2023, 21:19:31] [HomeConnect] Home Connect request #13: 200 OK +464ms
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] 7 of 8 programs are currently unavailable
>[05/11/2023, 21:19:31] [HomeConnect] Scheduling configuration schema write
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] "Program details" written to cache
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] Cache saved (21 entries)
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] Finished reading available program options
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] Adding services for 0 programs
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] Removing services for 13 programs
>[05/11/2023, 21:19:31] [HomeConnect] [Diskmaskin] Cache "Appliance commands" [sv-SE, updated 909 milliseconds ago]
>[05/11/2023, 21:19:34] [HomeConnect] Scheduled configuration schema write complete
>[05/11/2023, 21:20:22] [HomeConnect] [Diskmaskin] Event NOTIFY (1 items)
>[05/11/2023, 21:20:22] [HomeConnect] [Diskmaskin] BSH.Common.Option.RemainingProgramTime=12780 seconds (1 listeners)
>[05/11/2023, 21:20:22] [HomeConnect] [Diskmaskin] Program has 3 hours 33 minutes remaining
>[05/11/2023, 21:20:24] [HomeConnect] [Diskmaskin] Event STATUS (1 items)
>[05/11/2023, 21:20:24] [HomeConnect] [Diskmaskin] BSH.Common.Status.DoorState=BSH.Common.EnumType.DoorState.Open (1 listeners)
>[05/11/2023, 21:20:24] [HomeConnect] [Diskmaskin] Door open
>[05/11/2023, 21:20:31] [HomeConnect] [Diskmaskin] Event STATUS (1 items)
>[05/11/2023, 21:20:31] [HomeConnect] [Diskmaskin] BSH.Common.Status.DoorState=BSH.Common.EnumType.DoorState.Closed (1 listeners)
>[05/11/2023, 21:20:31] [HomeConnect] [Diskmaskin] Door closed
>[05/11/2023, 21:21:42] [HomeConnect] [Diskmaskin] Event NOTIFY (2 items)
>[05/11/2023, 21:21:42] [HomeConnect] [Diskmaskin] BSH.Common.Option.ProgramProgress=12 % (0 listeners)
>[05/11/2023, 21:21:42] [HomeConnect] [Diskmaskin] BSH.Common.Option.RemainingProgramTime=12720 seconds (1 listeners)
>[05/11/2023, 21:21:42] [HomeConnect] [Diskmaskin] Program has 3 hours 32 minutes remaining

**Comment by thoukydides** [comment]
That log all looks fine. As expected, the plugin does not issue any Home Connect API requests in response to the door being opened and closed, so does not affect the running programme. The configuration of which program switches are created should not affect that… and doesn’t in my own testing.

**Reply by fitur** [reply]
You are correct. I verified and all seems ok!
